src/debug/dram.c

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation.
+
+#include <rtos/kernel.h>
+#include <rtos/symbol.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+
+LOG_MODULE_REGISTER(dram_debug, CONFIG_SOF_LOG_LEVEL);
+
+static struct k_spinlock hot_path_lock;
+static unsigned int hot_path_depth;
+static const char *cold_path_fn;
+static bool hot_path_confirmed;
+
+void mem_cold_path_enter(const char *fn)
```

Comment:
We really need to put all debug logic behind a ```dbg_``` API prefix and a Kconfig to enable and disable.

Reply:
@lgirdwood so far we don't seem to have such a consistent convention. If you grep the source tree for `dbg_` you'll see a few internal uses and a few global macros in src/include/sof/debug/debug.h which don't seem to be used anywhere any more, so they can be removed. After that a consistent `dbg_*` namespace could be established. FWIW other files under src/debug/ don't use it either. One of the files there uses the `debug_` prefix

Reply:
Lets make it consistent starting with this file. We really have to namespace APIs better.